### PR TITLE
Add partialEval on Binary mlang

### PR DESCRIPTION
**Pull request by lchanmann**

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/mlang/mlang.js [hunk 1]
```diff
@@ -797,6 +797,32 @@ getArg2().prepareStatement(stmt);`
...
+        var newArg2 = this.arg2.partialEval();
+
+        if ( this.arg1 === newArg1 && this.arg2 === newArg2 ) return this;
+        return this.cls_.create({arg1: newArg1, arg2: newArg2});
+      },
+      javaCode: `
+        var newArg1 = getArg1().partialEval();
+        var newArg2 = getArg2().partialEval();
+
+        if ( getArg1().equals(newArg1) && getArg2().equals(newArg2) )
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js]: I think == would work here as well and be faster.

File: src/foam/mlang/mlang.js [hunk 2]
```diff
@@ -797,6 +797,32 @@ getArg2().prepareStatement(stmt);`
...
+          return this;
+
+        try {
+          var nu = (Binary) getClass().getDeclaredConstructor().newInstance();
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js:818]: Can't you just call newInstance() directly on the class?

File: src/foam/mlang/mlang.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/mlang/mlang.js]: Updated.

File: src/foam/mlang/mlang.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/mlang/mlang.js:818]: newInstance() is deprecated and replaced by clazz.getDeclaredConstructor().newInstance().

**Pull request merged**